File: CS/OsmSearchSample/MainForm.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraMap;

namespace OsmSearchSample {
    public partial class MainForm : Form {
        public MainForm() {
            InitializeComponent();
        }

        private void OnFormLoad(object sender, EventArgs e) {

            ImageLayer imageLayer = new ImageLayer();
            OpenStreetMapDataProvider dataProvider = new OpenStreetMapDataProvider();
            dataProvider.TileUriTemplate = "https://{0}.tile.openstreetmap.org/{1}/{2}/{3}.png";
            imageLayer.DataProvider = dataProvider;
            dataProvider.WebRequest += DataProvider_WebRequest;

            InformationLayer infoLayer = new InformationLayer();
            OsmSearchDataProvider osmSearchDataProvider = new OsmSearchDataProvider();
            osmSearchDataProvider.ResultsCount = 5;
            infoLayer.DataProvider = osmSearchDataProvider;

            mapControl.SearchPanelOptions.Width = 400;

            mapControl.Layers.AddRange(new LayerBase[] { imageLayer, infoLayer });
        }

        private void DataProvider_WebRequest(object sender, MapWebRequestEventArgs e) {
            e.UserAgent = "Sample app with OSM tiles and search";
            e.Referer = "https://www.mycompanysite.com/";
        }
    }
}

File: VB/OsmSearchSample/MainForm.vb
Imports DevExpress.XtraMap

Namespace OsmSearchSample
    Partial Public Class MainForm
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

#Region "#LayersSpecifying"
        Private Sub OnFormLoad(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            mapControl.Layers.AddRange(New LayerBase() {
                New ImageLayer With {
                    .DataProvider = New OpenStreetMapDataProvider With {
                        .TileUriTemplate = "YOUR_WEB_SERVICE.com/{1}/{2}/{3}"
                    }
                },
                New InformationLayer With {
                    .DataProvider = New OsmSearchDataProvider
                }
            })
        End Sub
#End Region ' #LayersSpecifying
    End Class
End Namespace